providers/amazon/src/airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -747,7 +748,22 @@ def conn(self) -> BaseAwsConnection:
 
     @property
     def async_conn(self):
+        """
+        [DEPRECATED] Get an aiobotocore client to use for async operations.
+
+        This property is deprecated. Accessing it in an async context will cause the event loop to block.
+        Use the async method `get_async_conn` instead.
+        """
```

Comment:
We should log a deprecation warning here as well.

Reply:
I originally did, but pre-commit complained about deprecation warnings in providers.
https://github.com/apache/airflow/pull/47326/commits/7c45860e474bda815275cc2eae30a566209eb9c7

Let me know how I can add it again without disappointing pre-commit.

Reply:
Done, let me know what you think.

Reply:
lgtm!